modules/ossm-installing-external-control-plane.adoc

```diff
@@ -0,0 +1,457 @@
+----
+
+.. Wait for the ingress gateway to be assigned an IP address by running the following command:
++
+[source,terminal]
+----
+$ oc --context "${CTX_CONTROL_PLANE_CLUSTER}" wait --for='jsonpath={.status.loadBalancer.ingress[].ip}' --timeout=300s svc istio-ingressgateway -n istio-system
+----
+
+.. Store the IP address of the ingress gateway in an environment variable by running the following command:
++
+[source,terminal]
+----
+$ export EXTERNAL_ISTIOD_ADDR=$(oc -n istio-system --context="${CTX_CONTROL_PLANE_CLUSTER}" get svc istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+----
+
+
+. Install {istio} on the data plane cluster:
+
+.. Create a project called `external-istiod` on the data plane cluster by running the following command:
++
+[source,terminal]
+----
+$ oc get project external-istiod --context "${CTX_DATA_PLANE_CLUSTER}" || oc new-project external-istiod --context "${CTX_DATA_PLANE_CLUSTER}"
+----
+
+.. Create an `Istio` resource on the data plane cluster by running the following command:
++
+[source,terminal]
+----
+$ cat <<EOF | oc --context "${CTX_DATA_PLANE_CLUSTER}" apply -f -
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: external-istiod
+spec:
+  version: v${ISTIO_VERSION}
+  namespace: external-istiod
+  profile: remote
+  values:
+    defaultRevision: external-istiod
+    global:
+      remotePilotAddress: ${EXTERNAL_ISTIOD_ADDR}
+      configCluster: true <1>
+    pilot:
+      configMap: true
+      istiodRemote:
+        injectionPath: /inject/cluster/cluster2/net/network1
+EOF
+----
+<1> This setting identifies the data plane cluster as the source of the mesh configuration.
+
+.. Create an `IstioCNI` resource on the data plane cluster by running the following command:
++
+[source,terminal]
+----
+$ cat <<EOF | oc --context "${CTX_DATA_PLANE_CLUSTER}" apply -f -
+apiVersion: sailoperator.io/v1
+kind: IstioCNI
+metadata:
+  name: default
+spec:
+  version: v${ISTIO_VERSION}
+  namespace: istio-cni
+EOF
+----
+
+. Set up the external {istio} control plane on the control plane cluster:
+
+.. Create a project called `external-istiod` on the control plane cluster by running the following command:
++
+[source,terminal]
+----
+$ oc get project external-istiod --context "${CTX_CONTROL_PLANE_CLUSTER}" || oc new-project external-istiod --context "${CTX_CONTROL_PLANE_CLUSTER}"
+----
+
+.. Create a `ServiceAccount` resource on the control plane cluster by running the following command:
++
+[source,terminal]
+----
+$ oc --context="${CTX_CONTROL_PLANE_CLUSTER}" create serviceaccount istiod-service-account -n external-istiod
+----
+
+.. Store the API server address for the data plane cluster in an environment variable by running the following command:
++
+[source,terminal]
+----
+$ DATA_PLANE_API_SERVER=https://<hostname_or_IP_address_of_the_API_server_for_the_data_plane_cluster>
```

Comment:
DATA_PLANE_API_SERVER_IP=$(oc get nodes -l node-role.kubernetes.io/control-plane --context "${CTX_CLUSTER2}" -o jsonpath='{.items[0].status.addresses[?(@.type == "InternalIP")].address}')

Reply:
I'm not sure this will work in general, since this produces the node's internal IP address. The reason we didn't provide a command to get the IP address was that we can't really know how the API server is exposed. For example, if there's a proxy in front of the API server, you would have to use the proxy's IP address here.

Reply:
DATA_PLANE_API_SERVER=https://<hostname_or_IP_address_of_the_API_server_for_the_data_plane_cluster>:6443